src/utils/BreakGlassCalldata.sol

```diff
@@ -0,0 +1,229 @@
+pragma solidity 0.8.19;
+
+import {ChainIds} from "@test/utils/ChainIds.sol";
+import {Addresses} from "@proposals/Addresses.sol";
+import {ITemporalGovernor} from "@protocol/Governance/ITemporalGovernor.sol";
+
+contract BreakGlassCalldata is ChainIds {
+    /// @notice address of the temporal governor
+    address[] public temporalGovernanceTargets;
+
+    /// @notice trusted senders for the temporal governor
+    ITemporalGovernor.TrustedSender[] public temporalGovernanceTrustedSenders;
+
+    /// @notice whitelisted calldata for the break glass guardian
+    bytes[] public approvedCalldata;
+
+    /// @notice whitelisted calldata for the temporal governor
+    bytes[] public temporalGovernanceCalldata;
+
+    struct Calls {
+        address target;
+        bytes call;
+    }
+
+    function buildWhitelistedCalldatas(
+        Addresses addresses
+    ) public returns (bytes memory) {
+        require(
+            temporalGovernanceTargets.length == 0,
+            "calldata already set in mip-18-c"
+        );
+        require(
+            temporalGovernanceTrustedSenders.length == 0,
+            "temporal gov trusted sender already set in mip-18-c"
+        );
+        require(
+            approvedCalldata.length == 0,
+            "approved calldata already set in mip-18-c"
+        );
+        require(
+            temporalGovernanceCalldata.length == 0,
+            "temporal gov calldata already set in mip-18-c"
+        );
+
+        address artemisTimelock = addresses.getAddress("MOONBEAM_TIMELOCK");
+        address temporalGovernor = addresses.getAddress(
+            "TEMPORAL_GOVERNOR",
+            sendingChainIdToReceivingChainId[block.chainid]
+        );
+
+        /// add temporal governor to list
+        temporalGovernanceTargets.push(temporalGovernor);
+
+        temporalGovernanceTrustedSenders.push(
+            ITemporalGovernor.TrustedSender({
+                chainId: moonBeamWormholeChainId, /// this chainId is 16 (moonBeamWormholeChainId) regardless of testnet or mainnet
+                addr: artemisTimelock /// this timelock on this chain
+            })
+        );
+
+        /// new break glass guardian call for adding artemis as an owner of the Temporal Governor
+
+        /// roll back trusted senders to artemis timelock
+        /// in reality this just adds the artemis timelock as a trusted sender
+        /// a second proposal is needed to revoke the Multichain Governor as a trusted sender
```

Comment:
Why can't we remove it from the same tx? I'm wondering, what is the issue with removing the current trusted sender after passing the restricted caller checks?